### ISSUE TITLE
Levenberg-Marquardt

### ISSUE THREAD

**Issue by vadixidav** [rust-cv/rust-cv#2]
Levenberg-Marquardt is so commonly used in computer vision for iterative optimization that it should have its own set of abstractions that allow it to be applied in different problem domains. This should go in its own crate. The crate should provide an implementation along with the appropriate abstractions to allow the user to plug in their implementation of the Jacobian matrix. It should also provide clear instructions on how to apply it to an optimization problem of multiple variables that a programmer can understand.

**Comment by mpizenberg** [comment]
In the direct visual odometry crate that I've been working on, I've implemented a LM optimization for the motion estimation step. But in order to get a useful trait abstraction I've built the trait for an iterative optimizer trait, and implemented it with a Levenberg Marquardt scheme also for 3 different small examples
* 1-dimensional regression,
* Rosenbrock function, a classic in optimization,
* an affine 2D image warping.

Those three examples each live in the examples directory of the crate: https://github.com/mpizenberg/visual-odometry-rs/tree/master/examples#optimization.

I thought I could share this as a reference for an implementation in Rust.
